resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala

```diff
@@ -393,5 +395,19 @@ package object config {
   private[yarn] val YARN_EXECUTOR_RESOURCE_TYPES_PREFIX = "spark.yarn.executor.resource."
   private[yarn] val YARN_DRIVER_RESOURCE_TYPES_PREFIX = "spark.yarn.driver.resource."
   private[yarn] val YARN_AM_RESOURCE_TYPES_PREFIX = "spark.yarn.am.resource."
-
+  lazy val IS_HADOOP_PROVIDED: Boolean = {
+    val configPath = "org/apache/spark/deploy/yarn/config.properties"
+    val propertyKey = "spark.yarn.isHadoopProvided"
+    try {
+      val prop = new Properties()
+      prop.load(ClassLoader.getSystemClassLoader.
+        getResourceAsStream(configPath))
+      prop.getProperty(propertyKey).toBoolean
+    } catch {
+      case e: Exception =>
+        log.warn(s"Can not load the default value of `$propertyKey` from " +
+          s"$configPath with error, ${e.toString}. Using false as a default value."
+        false
```

Comment:
Can we have `true` since it's the previous behavior?

Reply:
This variable naming is a little confusing because
`IS_HADOOP_PROVIDED` is used for `spark.yarn.populateHadoopClasspath`.

Reply:
If `spark.yarn.isHadoopProvided` doesn't exist, `IS_HADOOP_PROVIDED` becomes `false`. So, Hadoop is not provided. As a result, Spark needs to use the built-in Hadoop. But, this line seems to make `spark.yarn.populateHadoopClasspath=false`. IIRC, we need to populate hadoop class pathes if we are using built-in Hadoop, aren't we?

Reply:
When hadoop is provided, that means our Spark distribution doesn't have built-in hadoop; thus, we need to populate the hadoop classpath from Yarn. I can add comment there so people can understand it easier.

Reply:
But, what you mean is inconsistent with the PR title, `Only populate Hadoop classpath for no-hadoop build`. This PR is trying to set `IS_HADOOP_PROVIDED (false)` to `spark.yarn.populateHadoopClasspath` always for Hadoop distribution too, isn't it?